File: 001/ocaml/p001.ml
(*
  compile bytecode with  ocamlc q001.ml
  compile natively with  ocamlopt q001.ml

  run with ./a.out 1000
*)

let gauss_sum n = n*(n+1)/2

let gauss_sum_mod n m = (gauss_sum ((n - 1) / m)) * m;;

let n = int_of_string Sys.argv.(1) in
let sum3 = gauss_sum_mod n 3 in
let sum5 = gauss_sum_mod n 5 in
let sum15 = gauss_sum_mod n 15 in
let sum = (sum3 + sum5) - sum15 in
print_string ((string_of_int sum) ^ "\n")

File: 001/python/p001.py
#!/usr/bin/python -O 

import sys

def gauss_sum (n):
    return( (n * (n+1)) / 2)

def gauss_sum_mod (n, m):
    return (gauss_sum((n-1) / m) * m)

n = int(sys.argv[1])
sum3 = gauss_sum_mod(n, 3)
sum5 = gauss_sum_mod(n, 5)
sum15 = gauss_sum_mod(n, 15)
s = (sum3 + sum5) - sum15
print(s)
